### PR TITLE
Allow whitespace between square brackets in GeoJSON polygons

### LINKED ISSUE #5388: Geozone Geojson map rendering
### Description

[comment]: # (Feature or Bug? eg. Type: Bug)
*Type*: Bug

### Current Behavior
When importing a geojson file with spaces between leading brackets, the file imports correctly into a geozone but does not render on the budgets map if there are spaces between the leading square brackets.
This geojson doesn't display on map
image

This geojson displays correctly

image

The problem appears to be in the regex used for matching in the normalized_coordinates method which isn't allowing for whitespace

### Definition of Done
File is displayed when whitespace occurs between brackets

### PR DESCRIPTION
**Pull request by coslajohn**
Refactor regex in normalized_coordinates method to allow for whitespace between square brackets

## References

Closes #5388

### CODE REVIEW

**Comment by javierm** [comment]
@coslajohn This is great, thanks! :pray:

Here's a patch adding some tests and applying our whitespace conventions. Would it be possible for you to apply it?

```diff
diff --git a/app/models/geozone.rb b/app/models/geozone.rb
index eddf5a13f8..d3fc484d5a 100644
--- a/app/models/geozone.rb
+++ b/app/models/geozone.rb
@@ -30,8 +30,7 @@ class Geozone < ApplicationRecord
       if geojson.present?
         if geojson.match(/"coordinates"\s*:\s*\[\s*\[\s*\[\s*\[/)
           coordinates.reduce([], :concat).reduce([], :concat)
-         elsif geojson.match(/"coordinates"\s*:\s*\[\s*\[\s*\[/)
-  
+        elsif geojson.match(/"coordinates"\s*:\s*\[\s*\[\s*\[/)
           coordinates.reduce([], :concat)
         else
           coordinates
@@ -40,7 +39,7 @@ class Geozone < ApplicationRecord
         []
       end
     end
-    
+
     def coordinates
       JSON.parse(geojson)["geometry"]["coordinates"]
     end
diff --git a/spec/models/geozone_spec.rb b/spec/models/geozone_spec.rb
index c2b441f10b..973e611391 100644
--- a/spec/models/geozone_spec.rb
+++ b/spec/models/geozone_spec.rb
@@ -76,5 +76,77 @@ describe Geozone do
          [-3.9247799675785, 40.8789131852224]]
       )
     end
+
+    it "handles coordinates with three-dimensional arrays" do
+      geozone = build(:geozone, geojson: '{
+        "geometry": {
+          "type": "Polygon",
+          "coordinates": [[[40.8792937308316, -3.9259027239257],
+            [40.8788966596619, -3.9249047078766],
+            [40.8789131852224, -3.9247799675785]]]
+        }
+      }')
+
+      expect(geozone.outline_points).to eq(
+        [[-3.9259027239257, 40.8792937308316],
+         [-3.9249047078766, 40.8788966596619],
+         [-3.9247799675785, 40.8789131852224]]
+      )
+    end
+
+    it "handles coordinates with three-dimensional arrays with spaces between brackets" do
+      geozone = build(:geozone, geojson: '{
+        "geometry": {
+          "type": "Polygon",
+          "coordinates": [[
+            [40.8792937308316, -3.9259027239257],
+            [40.8788966596619, -3.9249047078766],
+            [40.8789131852224, -3.9247799675785]
+          ]]
+        }
+      }')
+
+      expect(geozone.outline_points).to eq(
+        [[-3.9259027239257, 40.8792937308316],
+         [-3.9249047078766, 40.8788966596619],
+         [-3.9247799675785, 40.8789131852224]]
+      )
+    end
+
+    it "handles coordinates with four-dimensional arrays" do
+      geozone = build(:geozone, geojson: '{
+        "geometry": {
+          "type": "Polygon",
+          "coordinates": [[[[40.8792937308316, -3.9259027239257],
+            [40.8788966596619, -3.9249047078766],
+            [40.8789131852224, -3.9247799675785]]]]
+        }
+      }')
+
+      expect(geozone.outline_points).to eq(
+        [[-3.9259027239257, 40.8792937308316],
+         [-3.9249047078766, 40.8788966596619],
+         [-3.9247799675785, 40.8789131852224]]
+      )
+    end
+
+    it "handles coordinates with four-dimensional arrays with spaces between brackets" do
+      geozone = build(:geozone, geojson: '{
+        "geometry": {
+          "type": "Polygon",
+          "coordinates": [[[
+            [40.8792937308316, -3.9259027239257],
+            [40.8788966596619, -3.9249047078766],
+            [40.8789131852224, -3.9247799675785]
+          ]]]
+        }
+      }')
+
+      expect(geozone.outline_points).to eq(
+        [[-3.9259027239257, 40.8792937308316],
+         [-3.9249047078766, 40.8788966596619],
+         [-3.9247799675785, 40.8789131852224]]
+      )
+    end
   end
 end

**Comment by coslajohn** [comment]
> @coslajohn This is great, thanks! 🙏
> 
> Here's a patch adding some tests and applying our whitespace conventions. Would it be possible for you to apply it?
> 

Patch applied now

**Comment by javierm** [comment]
@coslajohn Thanks a lot!
